### PR TITLE
Recovery fixes

### PR DESCRIPTION
**Pull request by ghedo**
Just a few small recovery fixes and improvements.

### CODE REVIEW

File: src/recovery.rs [hunk 1]
```diff
@@ -773,6 +779,418 @@ mod tests {
...
+                ""
+            ),
+            Ok(())
+        );
+
+        assert_eq!(r.sent[packet::EPOCH_APPLICATION].len(), 2);
+        assert_eq!(r.bytes_in_flight, 2000);
+        assert_eq!(r.lost_count, 0);
+
+        // Wait untill loss detection timer expires.
```
Comment (LPardue) [inline src/recovery.rs:886]:
```suggestion
        // Wait until loss detection timer expires.
```

File: src/recovery.rs [hunk 2]
```diff
@@ -773,6 +779,418 @@ mod tests {
...
+                ""
+            ),
+            Ok(())
+        );
+
+        assert_eq!(r.sent[packet::EPOCH_APPLICATION].len(), 1);
+        assert_eq!(r.bytes_in_flight, 1000);
+        assert_eq!(r.lost_count, 0);
+
+        // Wait untill loss detection timer expires.
```
Comment (LPardue) [inline src/recovery.rs:1059]:
```suggestion
        // Wait until loss detection timer expires.
```

File: src/recovery.rs (hunk 1 above)
Comment (ghedo) [reply src/recovery.rs]: Fixed, thanks.

File: src/recovery.rs (hunk 2 above)
Comment (ghedo) [reply src/recovery.rs]: Fixed, thanks.

File: src/recovery.rs [hunk 3]
```diff
@@ -779,6 +779,418 @@ mod tests {
...
+        let p = Sent {
+            pkt_num: 0,
+            frames: vec![],
+            time_sent: now,
+            size: 1000,
+            ack_eliciting: true,
+            in_flight: true,
+            delivered: 0,
+            delivered_time: std::time::Instant::now(),
+            recent_delivered_packet_sent_time: std::time::Instant::now(),
```
Comment (junhochoi) [inline src/recovery.rs:804]:
```suggestion
            delivered_time: now,
            recent_delivered_packet_sent_time: now,
```

There are many similar lines below as well.

File: src/recovery.rs [hunk 4]
```diff
@@ -779,6 +779,418 @@ mod tests {
...
```
Comment (junhochoi) [inline src/recovery.rs:976]:
```suggestion
            delivered_time: now,
            recent_delivered_packet_sent_time: now,
```

File: src/recovery.rs (hunk 3 above)
Comment (ghedo) [reply src/recovery.rs]: Ah yes. Fixed now, thanks.

**Pull request merged**